src/Parser.js

```diff
@@ -815,43 +815,51 @@ export default class Parser {
     parseGroup(optional?: boolean, mode?: Mode): ?ParsedFuncOrArg {
         const outerMode = this.mode;
         const firstToken = this.nextToken;
-        // Try to parse an open brace
+        let semisimple;
+        let groupEnd;
+        // Try to parse an open brace or \begingroup
         if (this.nextToken.text === (optional ? "[" : "{")) {
-            // Switch to specified mode before we expand symbol after brace
-            if (mode) {
-                this.switchMode(mode);
-            }
-            // Start a new group namespace
-            this.gullet.beginGroup();
-            // If we get a brace, parse an expression
-            this.consume();
-            const expression = this.parseExpression(false, optional ? "]" : "}");
-            const lastToken = this.nextToken;
-            // Switch mode back before consuming symbol after close brace
-            if (mode) {
-                this.switchMode(outerMode);
-            }
-            // End group namespace before consuming symbol after close brace
-            this.gullet.endGroup();
-            // Make sure we get a close brace
-            this.expect(optional ? "]" : "}");
-            return newArgument({
-                type: "ordgroup",
-                mode: this.mode,
-                loc: SourceLocation.range(firstToken, lastToken),
-                body: expression,
-            }, firstToken.range(lastToken, firstToken.text));
-        } else {
-            // Otherwise, just return a nucleus, or nothing for an optional group
-            if (mode) {
-                this.switchMode(mode);
-            }
+            groupEnd = optional ? "]" : "}";
+        } else if (!optional && this.nextToken.text === "\\begingroup") {
+            // A group formed by /begingroup...\endgroup is a semi-simple group,
+            // which doesn't affect spacing in math mode, i.e., is transparent
+            semisimple = true;
```

Comment:
Can you add a reference to the TeXBook or similar resource?

Reply:
Did you determine the behavior by seeing what LaTeX does for different test cases?

Reply:
FWIW, this behavior matches my experience testing with LaTeX. I don't see it documented in The TeXbook. But the TeX source code definitely calls groups started with `\begingroup` "semi-simple groups" (search for `begin_group`), and it definitely does something special with them (search for `semi_simple_group`). But I'm no expert at that code...

Reply:
@kevinbarabash I've added a link to stack exchange answer, as it has enough up-votes.